test/unit_tests/cellular/cellular.cpp

```diff
@@ -78,6 +99,64 @@ TEST_CASE("cellular_signal()") {
         }
     }
 
+    SECTION("CellularSignalHal RSSI and QUAL") {
+        NetStatus status;
+        memset(&status, 0, sizeof(status));
+        status.rssi = -50; // arbitrary non-zero values
+        status.qual = 37;  //  "
+
+        SECTION("CellularSignalHal::rssi and CellularSignalHal::qual are set to expected values") {
+            CellularSignalHal signal;
+            memset(&signal, 0, sizeof(signal));
+            REQUIRE(cellular_signal_impl(&signal, nullptr, true, status) == SYSTEM_ERROR_NONE);
+            REQUIRE(signal.rssi == -50);
+            REQUIRE(signal.qual == 37);
+        }
+    }
+
+    SECTION("UNKNOWN") {
+        NetStatus status;
+        memset(&status, 0, sizeof(status));
+        status.act = ACT_UNKNOWN;
+
+        SECTION("values reported by ACT_UNKNOWN case") {
+            status.rxlev = 10; // seed all values as 10 just to ensure default case is hit
+            status.rxqual = 10;
+            status.rscp = 10;
+            status.ecno = 10;
+            status.rsrp = 10;
+            status.rsrq = 10;
+
+            cellular_signal_t sig = {0};
+            sig.size = sizeof(sig);
+            REQUIRE(cellular_signal_impl(nullptr, &sig, true, status) == SYSTEM_ERROR_UNKNOWN);
+            REQUIRE(sig.rat == NET_ACCESS_TECHNOLOGY_NONE);
+            REQUIRE(sig.strength == 0);
+            REQUIRE(sig.quality == 0);
+            REQUIRE(sig.rssi == std::numeric_limits<int32_t>::min());
+            REQUIRE(sig.qual == std::numeric_limits<int32_t>::min());
+
+            SECTION("CellularSignal") {
+                CellularSignal cs;
+                REQUIRE(cs.fromHalCellularSignal(sig) == true);
+                REQUIRE(cs.getAccessTechnology() == NET_ACCESS_TECHNOLOGY_NONE);
+                REQUIRE(cs.getStrength() < 0.0f);
+                REQUIRE(cs.getStrengthValue() == 0.0f);
+                REQUIRE(cs.getQuality() < 0.0f);
+                REQUIRE(cs.getQualityValue() == 0.0f);
+            }
+
+            SECTION("CellularSignal") {
```

Comment:
See previous comment. This feels like `SECTION` abuse.

Reply:
When using TEST_CASE and SECTION style, there is no section abuse... that's just how you use SECTION s.